streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -108,7 +112,7 @@
 
             // add the operation metrics with additional tags
             final Map<String, String> tagMap = metrics.tagMap("task-id", taskName);
-            taskCommitTimeSensor = metrics.taskLevelSensor("commit", taskName, Sensor.RecordingLevel.DEBUG, parent);
+            taskCommitTimeSensor = metrics.taskLevelSensor(taskName, "commit", Sensor.RecordingLevel.DEBUG, parent);
```

Comment:
Can you open PRs for older branches to back-port this fix?

Reply:
FWIW, it wouldn't break anything. This is only the name of the sensor, which just needs to be unique, and it's just as unique in this order, unless some other task is named "commit" (which wouldn't happen).

Reply:
To be clear, we should fix it, just not sure if we need to bother backporting.

Reply:
Cannot follow the argument? It seems to be an easy fix and we should back port all fixes by default and only not back port if there is a good reason for not doing it.

@guozhangwang Is there already a PR or ticket so this is not dropped on the floor?

Reply:
@vvcephei I think it did break sth.. the sensorName is used as part of `fullSensorName = key + "." + sensorName;`. With `sensorName == taskName`, we will not add them since `Sensor s = getSensor(name);` already exists, i.e. we will lose some metrics, right?

Reply:
@mjsax fair enough!

@guozhangwang I think you're right. The full sensor name is still unique, but it would mess up the ability to unload sensors (we basically wouldn't be able to unload any task level sensors). My bad.

Reply:
Maybe create (internal) SensorName type (alias for String) so that there is no chance of mixup in the future.